src/activity_actor.cpp

```diff
@@ -1754,10 +1754,20 @@ void craft_activity_actor::do_turn( player_activity &act, Character &crafter )
     // This is to ensure we don't over count skill steps
     craft.item_counter = std::min( craft.item_counter, 10'000'000 );
 
-    // Skill and tools are gained/consumed after every 5% progress
+    // This nominal craft time is also how many practice ticks to perform
+    // spread out evenly across the actual duration.
+    const float total_practice_ticks = rec.time_to_craft_moves( crafter,
```

Comment:
rvalue here and the division next line are both of `double` type.
```suggestion
    const double total_practice_ticks = rec.time_to_craft_moves( crafter,
```